R/FilterSpectroOut.R

```diff
@@ -0,0 +1,659 @@
+                                     nameProteotypic="True",
+                                     filterMissedCleaved=FALSE,
+                                     maxMissedCleave=2,
+                                     infoMissedCleave="NMissedCleavages"){
+    ## check input and data
+    ## check mode setting
+    if(!tolower(mode) %in% c("default", "htonly", "fthtjoin", "liponly")){
+        stop("Please set 'mode' to one of the allowed options: c('default',
+ 'HTonly', 'FTHTjoin', 'LiPonly'.")
+    }
+
+    ## check quantityList and quantityMatrix input
+    if(is.null(quantityList)){
+        if(is.null(quantityMatrix)){
+            stop("Please provide an input with peptide or protein quantities
+using 'quantityList' or 'quantityMatrix'")
+        }
+        if(tolower(mode) != 'default'){
+            stop("'quantityList' can only be filtered if mode is set to default.
+Either set mode='default' or provide quantityList as input.")
+        }
+        else{
+            quantityList <- as.list(quantityMatrix)
+        }
+    }
+
+    else if(!is.null(quantityMatrix)){
+        message("'quantityList' and 'quantitMatrix' are provided. Only using
+'quantityList'.")
+    }
+
+    ## check if annotPP is required and provided
+    if((tolower(mode) %in% c("htonly", "fthtjoin")|filterTryptic!="none"|
+       filterProteotryptic|filterMissedCleaved)&is.null(annotPP)){
+        stop("Please provide annotPP.")
+    }
+
+    ## check input to filterTryptic
+    if(!(is.logical(filterTryptic)&length(filterTryptic)==1)){
+        stop("'filterTryptic' is not set correctly. Please set it to either
+'TRUE' or 'FALSE'.")
+    }
+
+    ## check input filterNA
+    if(!tolower(filterNA) %in% c("all", "bycondition", "none")){
+        stop("Please set 'filterNA' to one of the allowed options: c('all',
+'byCondition', 'none'.")
+    }
+
+    if(tolower(filterNA) == 'bycondition'&is.null(annotS)){
+        stop("Please provide annotS for filtering NAs by conidition.")
+    }
+
+    message("Preprocessing quantityList.")
+
+    ## removing TrPPep and mapping TrPProt to HT peptides in case the function
+    ## is run in 'runHT' mode
+    if(tolower(mode) == "fthtjoin"){
+        if(filterTryptic){
+            message("'mode' is set to 'FTHTjoin', setting 'filterTryptic' to
+'FALSE'.")
+            filterTryptic <- FALSE
+        }
+
+        if(paste(names(quantityList), collapse = "") ==
+           c("LiPPepTrPPepTrPProt")){
+            quantityList <- filterForFTHTjoin(quantityList, annotPP,
+                                              nameIDQuant, nameProtQuant)
+        }
+
+        else if(paste(names(quantityList), collapse = "") != c("LiPPepLiPProt")){
+            stop("Function is run in 'FTHTjoin' mode and names of the
+quantityList do not meet the expectations. Please provide either a list with
+matrices defined as 'LiPPep', 'TrPPep' and 'TrPProt' or a list with two
+matrices defined as 'LiPPep' and 'LiPProt'.")
+        }
+    }
```

Comment:
partially check the list entries here anyway if you check the correct expected data types maybe move that up to that block as well